boa_engine/src/error.rs

```diff
@@ -940,9 +940,7 @@ impl JsNativeError {
             JsNativeErrorKind::Uri => (constructors.uri_error().prototype(), ErrorKind::Uri),
             #[cfg(feature = "fuzz")]
             JsNativeErrorKind::NoInstructionsRemain => {
-                unreachable!(
-                    "The NoInstructionsRemain native error cannot be converted to an opaque type."
-                )
+                (constructors.eval_error().prototype(), ErrorKind::Eval)
```

Comment:
Why do we need to convert the native error into an opaque one? My concern would be that this error could be caught using `try`-`catch`

Reply:
If the error is thrown async then boa will convert it into an opaque error when it rejects the promise. This causes a panic.  If an out of gas error is caught it will immediately be rethrown because you're still out of gas.